app/values/validated_property.rb

```diff
@@ -0,0 +1,13 @@
+class ValidatedProperty
+  vattr_initialize :validated_property, :property_accessor
+end
+
+module Kernel
```

Comment:
Instead of Kernel, consider making a separate module "Conversions" or some such thing. Then you would include that conversion explicitly in the class that uses the ValidatedProperty.

Also might I suggest taking a look at my PowerConverter gem?